test/e2e/prometheusagent_test.go

```diff
@@ -183,3 +188,81 @@ func testPrometheusAgentStatusScale(t *testing.T) {
 
 	require.Equal(t, int32(2), pAgent.Status.Shards)
 }
+
+func testPromAgentDaemonSetResourceUpdate(t *testing.T) {
+	//t.Parallel()
+	ctx := context.Background()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	ns := framework.CreateNamespace(ctx, t, testCtx)
+	framework.SetupPrometheusRBAC(ctx, t, testCtx, ns)
+	_, err := framework.CreateOrUpdatePrometheusOperatorWithOpts(
+		ctx, testFramework.PrometheusOperatorOpts{
+			Namespace:           ns,
+			AllowedNamespaces:   []string{ns},
+			EnabledFeatureGates: []string{"PrometheusAgentDaemonSet"},
+		},
+	)
+	require.NoError(t, err)
+
+	name := "test"
+	p := framework.MakeBasicPrometheusAgentDaemonSet(ns, name)
+
+	p.Spec.Resources = v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+	}
+
+	p, err = framework.CreatePrometheusAgentAndWaitUntilReady(context.Background(), ns, p)
+	require.NoError(t, err)
+
+	pods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, pa.ListOptions(name))
+	require.NoError(t, err)
+
+	res := pods.Items[0].Spec.Containers[0].Resources
+	require.Equal(t, res, p.Spec.Resources)
+
+	p, err = framework.PatchPrometheusAgentAndWaitUntilReady(
+		context.Background(),
+		p.Name,
+		ns,
+		monitoringv1alpha1.PrometheusAgentSpec{
+			Mode: ptr.To("DaemonSet"),
+			CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+
+	var pollErr error
+	err = wait.PollUntilContextTimeout(context.Background(), 30*time.Second, 30*time.Minute, false, func(ctx context.Context) (bool, error) {
+		pods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, pa.ListOptions(name))
+		if err != nil {
+			pollErr = fmt.Errorf("failed to list Prometheus Agent DaemonSet pods: %w", err)
+			return false, nil
+		}
+
+		if len(pods.Items) != 1 {
+			pollErr = fmt.Errorf("expected 1 pod, actual number of pods: %d", len(pods.Items))
```

Comment:
@simonpasquier @ArthurSens The test is failing in this line: `expected 1 pod, actual number of pods: 2`

I'll investigate later today or tomorrow. If you have time pls investigate too!

As I said here, the test follow a similar Prometheus test, to check if Prometheus Agent can be updated: https://github.com/prometheus-operator/prometheus-operator/pull/6787#discussion_r1706605187

Reply:
Actually now I realize. We're running the test on 2 nodes, so we should actually expect 2 pods, right? :thinking:

Reply:
See my first comment: it's fine to check the daemonset definition rather than the pods.